samples/crypto/ecjpake/src/main.c

```diff
@@ -0,0 +1,261 @@
+	size_t outlen;
+	psa_status_t status;
+
+	/* Server provides data to client */
+	for (uint32_t i = 0; i < ARRAY_SIZE(round1_steps); i++) {
+		status = psa_pake_output(server, round1_steps[i], buffer, sizeof(buffer), &outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_output failed. (Error: %d)", status);
+			return status;
+		}
+
+		status = psa_pake_input(client, round1_steps[i], buffer, outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_input failed. (Error: %d)", status);
+			return status;
+		}
+	}
+
+	/* Client provides data to server. */
+	for (uint32_t i = 0; i < ARRAY_SIZE(round1_steps); i++) {
+		status = psa_pake_output(client, round1_steps[i], buffer, sizeof(buffer), &outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_output failed. (Error: %d)", status);
+			return status;
+		}
+
+		status = psa_pake_input(server, round1_steps[i], buffer, outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_input failed. (Error: %d)", status);
+			return status;
+		}
+	}
+
+	psa_pake_step_t round2_steps[] = { PSA_PAKE_STEP_KEY_SHARE, PSA_PAKE_STEP_ZK_PUBLIC,
+					   PSA_PAKE_STEP_ZK_PROOF };
+
+	/* Server provides data to client */
+	for (uint32_t i = 0; i < ARRAY_SIZE(round2_steps); i++) {
+		status = psa_pake_output(server, round2_steps[i], buffer, sizeof(buffer), &outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_output failed. (Error: %d)", status);
+			return status;
+		}
+
+		status = psa_pake_input(client, round2_steps[i], buffer, outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_input failed. (Error: %d)", status);
+			return status;
+		}
+	}
+
+	/* Client provides data to server. */
+	for (uint32_t i = 0; i < ARRAY_SIZE(round2_steps); i++) {
+		status = psa_pake_output(client, round2_steps[i], buffer, sizeof(buffer), &outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_output failed. (Error: %d)", status);
+			return status;
+		}
+
+		status = psa_pake_input(server, round2_steps[i], buffer, outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_input failed. (Error: %d)", status);
+			return status;
+		}
+	}
+
+	return status;
+}
+
+int main(void)
+{
+	psa_status_t status = psa_crypto_init();
+
+	if (status != PSA_SUCCESS) {
+		LOG_INF("psa_crypto_init failed. (Error: %d)", status);
+	}
+
+	psa_pake_cipher_suite_t cipher_suite = PSA_PAKE_CIPHER_SUITE_INIT;
+
+	psa_pake_cs_set_algorithm(&cipher_suite, PSA_ALG_JPAKE);
+	psa_pake_cs_set_primitive(&cipher_suite, PSA_PAKE_PRIMITIVE(PSA_PAKE_PRIMITIVE_TYPE_ECC,
+								    PSA_ECC_FAMILY_SECP_R1, 256));
+	psa_pake_cs_set_hash(&cipher_suite, PSA_ALG_SHA_256);
+
+	psa_key_attributes_t key_attributes = PSA_KEY_ATTRIBUTES_INIT;
+
+	psa_set_key_usage_flags(&key_attributes, PSA_KEY_USAGE_DERIVE);
+	psa_set_key_algorithm(&key_attributes, PSA_ALG_JPAKE);
+	psa_set_key_type(&key_attributes, PSA_KEY_TYPE_PASSWORD);
+
+	psa_key_id_t key;
+
+	status = psa_import_key(&key_attributes, "password", 8, &key);
```

Comment:
The 8 here and the 6 afterwards are  not easy to understand. Probably it's more clear to add the names as global const uint8_t and then use the sizeof() for this.

Reply:
Really? 

sizeof will include the null terminator so that can be surprising... So I can do strlen or sizeof - 1?